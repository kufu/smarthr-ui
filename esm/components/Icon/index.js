export { Icon, iconMap } from './Icon';
//# sourceMappingURL=index.js.map