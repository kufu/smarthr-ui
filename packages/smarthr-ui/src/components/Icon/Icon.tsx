import {
  FaAddressBook,
  FaAddressCard,
  FaAngleDown,
  FaAngleLeft,
  FaAngleRight,
  FaAngleUp,
  FaAnglesLeft,
  FaAnglesRight,
  FaArrowDown,
  FaArrowDownWideShort,
  FaArrowLeft,
  FaArrowRight,
  FaArrowRightArrowLeft,
  FaArrowRotateLeft,
  FaArrowRotateRight,
  FaArrowUp,
  FaArrowUp19,
  FaArrowUpWideShort,
  FaArrowsRotate,
  FaAsterisk,
  FaBabyCarriage,
  FaBan,
  FaBars,
  FaBarsStaggered,
  FaBell,
  FaBellSlash,
  FaBolt,
  FaBook,
  FaBookOpen,
  FaBoxArchive,
  FaBuilding,
  FaBullhorn,
  FaCakeCandles,
  FaCalculator,
  FaCalendarCheck,
  FaCalendarDays,
  FaCalendarXmark,
  FaCaretDown,
  FaCaretLeft,
  FaCaretRight,
  FaCaretUp,
  FaChartArea,
  FaChartBar,
  FaChartColumn,
  FaChartLine,
  FaChartPie,
  FaCheck,
  FaChevronDown,
  FaChevronLeft,
  FaChevronRight,
  FaCircle,
  FaCircleCheck,
  FaCircleDot,
  FaCircleExclamation,
  FaCircleInfo,
  FaCircleMinus,
  FaCirclePlus,
  FaCircleQuestion,
  FaCircleUser,
  FaCircleXmark,
  FaClipboard,
  FaClipboardCheck,
  FaClipboardList,
  FaClock,
  FaClockRotateLeft,
  FaCloudArrowDown,
  FaCloudArrowUp,
  FaCode,
  FaComment,
  FaCommentDots,
  FaComments,
  FaCompress,
  FaCopy,
  FaCube,
  FaCubes,
  FaDatabase,
  FaDoorOpen,
  FaDownLeftAndUpRightToCenter,
  FaEllipsis,
  FaEllipsisVertical,
  FaEnvelope,
  FaEnvelopeOpen,
  FaExpand,
  FaEye,
  FaEyeSlash,
  FaFile,
  FaFileArrowDown,
  FaFileCode,
  FaFileCsv,
  FaFileExcel,
  FaFileExport,
  FaFileLines,
  FaFilePdf,
  FaFilePowerpoint,
  FaFileWord,
  FaFilter,
  FaFloppyDisk,
  FaFolder,
  FaFolderOpen,
  FaFolderPlus,
  FaFont,
  FaGauge,
  FaGear,
  FaGears,
  FaGlobe,
  FaGraduationCap,
  FaGrip,
  FaGripLines,
  FaGripLinesVertical,
  FaGripVertical,
  FaHourglassHalf,
  FaHouse,
  FaIdBadge,
  FaIdCard,
  FaImage,
  FaInbox,
  FaIntercom,
  FaKey,
  FaLeftLong,
  FaLeftRight,
  FaLightbulb,
  FaLink,
  FaList,
  FaListOl,
  FaListUl,
  FaLock,
  FaLockOpen,
  FaMagnifyingGlass,
  FaMagnifyingGlassMinus,
  FaMagnifyingGlassPlus,
  FaMessage,
  FaMinus,
  FaMoneyCheckDollar,
  FaPaintbrush,
  FaPaperPlane,
  FaPaperclip,
  FaPen,
  FaPenToSquare,
  FaPencil,
  FaPiggyBank,
  FaPlus,
  FaPowerOff,
  FaPrint,
  FaRegCircleQuestion,
  FaRegLightbulb,
  FaReply,
  FaRightFromBracket,
  FaRightLeft,
  FaRightLong,
  FaRightToBracket,
  FaRobot,
  FaRocket,
  FaRotate,
  FaRotateLeft,
  FaShareNodes,
  FaShieldHalved,
  FaSliders,
  FaSort,
  FaSortDown,
  FaSortUp,
  FaSquareCheck,
  FaSquareMinus,
  FaSquarePlus,
  FaSquarePollVertical,
  FaStar,
  FaSuitcaseMedical,
  FaTable,
  FaTableColumns,
  FaTableList,
  FaTag,
  FaTags,
  FaThumbtack,
  FaToolbox,
  FaTrash,
  FaTrashCan,
  FaTriangleExclamation,
  FaTruck,
  FaTruckRampBox,
  FaUpDown,
  FaUpRightAndDownLeftFromCenter,
  FaUpRightFromSquare,
  FaUser,
  FaUserCheck,
  FaUserClock,
  FaUserLarge,
  FaUserPen,
  FaUserPlus,
  FaUsers,
  FaWindowMaximize,
  FaWindowMinimize,
  FaWindowRestore,
  FaXmark,
  FaYenSign,
} from 'react-icons/fa6'

import { createIcon } from './generateIcon'

export const FaAddressBookIcon = /*#__PURE__*/ createIcon(FaAddressBook)
export const FaAddressCardIcon = /*#__PURE__*/ createIcon(FaAddressCard)
export const FaAngleDownIcon = /*#__PURE__*/ createIcon(FaAngleDown)
export const FaAngleLeftIcon = /*#__PURE__*/ createIcon(FaAngleLeft)
export const FaAngleRightIcon = /*#__PURE__*/ createIcon(FaAngleRight)
export const FaAngleUpIcon = /*#__PURE__*/ createIcon(FaAngleUp)
export const FaAnglesLeftIcon = /*#__PURE__*/ createIcon(FaAnglesLeft)
export const FaAnglesRightIcon = /*#__PURE__*/ createIcon(FaAnglesRight)
export const FaArrowDownIcon = /*#__PURE__*/ createIcon(FaArrowDown)
export const FaArrowDownWideShortIcon = /*#__PURE__*/ createIcon(FaArrowDownWideShort)
export const FaArrowLeftIcon = /*#__PURE__*/ createIcon(FaArrowLeft)
export const FaArrowRightIcon = /*#__PURE__*/ createIcon(FaArrowRight)
export const FaArrowRightArrowLeftIcon = /*#__PURE__*/ createIcon(FaArrowRightArrowLeft)
export const FaArrowRotateLeftIcon = /*#__PURE__*/ createIcon(FaArrowRotateLeft)
export const FaArrowRotateRightIcon = /*#__PURE__*/ createIcon(FaArrowRotateRight)
export const FaArrowUp19Icon = /*#__PURE__*/ createIcon(FaArrowUp19)
export const FaArrowUpIcon = /*#__PURE__*/ createIcon(FaArrowUp)
export const FaArrowUpWideShortIcon = /*#__PURE__*/ createIcon(FaArrowUpWideShort)
export const FaArrowsRotateIcon = /*#__PURE__*/ createIcon(FaArrowsRotate)
export const FaAsteriskIcon = /*#__PURE__*/ createIcon(FaAsterisk)
export const FaBabyCarriageIcon = /*#__PURE__*/ createIcon(FaBabyCarriage)
export const FaBanIcon = /*#__PURE__*/ createIcon(FaBan)
export const FaBarsIcon = /*#__PURE__*/ createIcon(FaBars)
export const FaBarsStaggeredIcon = /*#__PURE__*/ createIcon(FaBarsStaggered)
export const FaBellIcon = /*#__PURE__*/ createIcon(FaBell)
export const FaBellSlashIcon = /*#__PURE__*/ createIcon(FaBellSlash)
export const FaBoltIcon = /*#__PURE__*/ createIcon(FaBolt)
export const FaBookIcon = /*#__PURE__*/ createIcon(FaBook)
export const FaBookOpenIcon = /*#__PURE__*/ createIcon(FaBookOpen)
export const FaBoxArchiveIcon = /*#__PURE__*/ createIcon(FaBoxArchive)
export const FaBuildingIcon = /*#__PURE__*/ createIcon(FaBuilding)
export const FaBullhornIcon = /*#__PURE__*/ createIcon(FaBullhorn)
export const FaCakeCandlesIcon = /*#__PURE__*/ createIcon(FaCakeCandles)
export const FaCalculatorIcon = /*#__PURE__*/ createIcon(FaCalculator)
export const FaCalendarCheckIcon = /*#__PURE__*/ createIcon(FaCalendarCheck)
export const FaCalendarDaysIcon = /*#__PURE__*/ createIcon(FaCalendarDays)
export const FaCalendarXmarkIcon = /*#__PURE__*/ createIcon(FaCalendarXmark)
export const FaCaretDownIcon = /*#__PURE__*/ createIcon(FaCaretDown)
export const FaCaretLeftIcon = /*#__PURE__*/ createIcon(FaCaretLeft)
export const FaCaretRightIcon = /*#__PURE__*/ createIcon(FaCaretRight)
export const FaCaretUpIcon = /*#__PURE__*/ createIcon(FaCaretUp)
export const FaChartAreaIcon = /*#__PURE__*/ createIcon(FaChartArea)
export const FaChartBarIcon = /*#__PURE__*/ createIcon(FaChartBar)
export const FaChartColumnIcon = /*#__PURE__*/ createIcon(FaChartColumn)
export const FaChartLineIcon = /*#__PURE__*/ createIcon(FaChartLine)
export const FaChartPieIcon = /*#__PURE__*/ createIcon(FaChartPie)
export const FaCheckIcon = /*#__PURE__*/ createIcon(FaCheck)
export const FaChevronDownIcon = /*#__PURE__*/ createIcon(FaChevronDown)
export const FaChevronLeftIcon = /*#__PURE__*/ createIcon(FaChevronLeft)
export const FaChevronRightIcon = /*#__PURE__*/ createIcon(FaChevronRight)
export const FaCircleCheckIcon = /*#__PURE__*/ createIcon(FaCircleCheck)
export const FaCircleDotIcon = /*#__PURE__*/ createIcon(FaCircleDot)
export const FaCircleExclamationIcon = /*#__PURE__*/ createIcon(FaCircleExclamation)
export const FaCircleIcon = /*#__PURE__*/ createIcon(FaCircle)
export const FaCircleInfoIcon = /*#__PURE__*/ createIcon(FaCircleInfo)
export const FaCircleMinusIcon = /*#__PURE__*/ createIcon(FaCircleMinus)
export const FaCirclePlusIcon = /*#__PURE__*/ createIcon(FaCirclePlus)
export const FaCircleQuestionIcon = /*#__PURE__*/ createIcon(FaCircleQuestion)
export const FaCircleUserIcon = /*#__PURE__*/ createIcon(FaCircleUser)
export const FaCircleXmarkIcon = /*#__PURE__*/ createIcon(FaCircleXmark)
export const FaClipboardCheckIcon = /*#__PURE__*/ createIcon(FaClipboardCheck)
export const FaClipboardIcon = /*#__PURE__*/ createIcon(FaClipboard)
export const FaClipboardListIcon = /*#__PURE__*/ createIcon(FaClipboardList)
export const FaClockIcon = /*#__PURE__*/ createIcon(FaClock)
export const FaClockRotateLeftIcon = /*#__PURE__*/ createIcon(FaClockRotateLeft)
export const FaCloudArrowDownIcon = /*#__PURE__*/ createIcon(FaCloudArrowDown)
export const FaCloudArrowUpIcon = /*#__PURE__*/ createIcon(FaCloudArrowUp)
export const FaCodeIcon = /*#__PURE__*/ createIcon(FaCode)
export const FaCommentDotsIcon = /*#__PURE__*/ createIcon(FaCommentDots)
export const FaCommentIcon = /*#__PURE__*/ createIcon(FaComment)
export const FaCommentsIcon = /*#__PURE__*/ createIcon(FaComments)
export const FaCompressIcon = /*#__PURE__*/ createIcon(FaCompress)
export const FaCopyIcon = /*#__PURE__*/ createIcon(FaCopy)
export const FaCubeIcon = /*#__PURE__*/ createIcon(FaCube)
export const FaCubesIcon = /*#__PURE__*/ createIcon(FaCubes)
export const FaDatabaseIcon = /*#__PURE__*/ createIcon(FaDatabase)
export const FaDoorOpenIcon = /*#__PURE__*/ createIcon(FaDoorOpen)
export const FaDownLeftAndUpRightToCenterIcon = /*#__PURE__*/ createIcon(
  FaDownLeftAndUpRightToCenter,
)
export const FaEllipsisIcon = /*#__PURE__*/ createIcon(FaEllipsis)
export const FaEllipsisVerticalIcon = /*#__PURE__*/ createIcon(FaEllipsisVertical)
export const FaEnvelopeIcon = /*#__PURE__*/ createIcon(FaEnvelope)
export const FaEnvelopeOpenIcon = /*#__PURE__*/ createIcon(FaEnvelopeOpen)
export const FaExpandIcon = /*#__PURE__*/ createIcon(FaExpand)
export const FaEyeIcon = /*#__PURE__*/ createIcon(FaEye)
export const FaEyeSlashIcon = /*#__PURE__*/ createIcon(FaEyeSlash)
export const FaFileArrowDownIcon = /*#__PURE__*/ createIcon(FaFileArrowDown)
export const FaFileCodeIcon = /*#__PURE__*/ createIcon(FaFileCode)
export const FaFileCsvIcon = /*#__PURE__*/ createIcon(FaFileCsv)
export const FaFileExcelIcon = /*#__PURE__*/ createIcon(FaFileExcel)
export const FaFileExportIcon = /*#__PURE__*/ createIcon(FaFileExport)
export const FaFileIcon = /*#__PURE__*/ createIcon(FaFile)
export const FaFileLinesIcon = /*#__PURE__*/ createIcon(FaFileLines)
export const FaFilePdfIcon = /*#__PURE__*/ createIcon(FaFilePdf)
export const FaFilePowerpointIcon = /*#__PURE__*/ createIcon(FaFilePowerpoint)
export const FaFileWordIcon = /*#__PURE__*/ createIcon(FaFileWord)
export const FaFilterIcon = /*#__PURE__*/ createIcon(FaFilter)
export const FaFloppyDiskIcon = /*#__PURE__*/ createIcon(FaFloppyDisk)
export const FaFolderIcon = /*#__PURE__*/ createIcon(FaFolder)
export const FaFolderOpenIcon = /*#__PURE__*/ createIcon(FaFolderOpen)
export const FaFolderPlusIcon = /*#__PURE__*/ createIcon(FaFolderPlus)
export const FaFontIcon = /*#__PURE__*/ createIcon(FaFont)
export const FaGaugeIcon = /*#__PURE__*/ createIcon(FaGauge)
export const FaGearIcon = /*#__PURE__*/ createIcon(FaGear)
export const FaGearsIcon = /*#__PURE__*/ createIcon(FaGears)
export const FaGlobeIcon = /*#__PURE__*/ createIcon(FaGlobe)
export const FaGraduationCapIcon = /*#__PURE__*/ createIcon(FaGraduationCap)
export const FaGripIcon = /*#__PURE__*/ createIcon(FaGrip)
export const FaGripLinesIcon = /*#__PURE__*/ createIcon(FaGripLines)
export const FaGripLinesVerticalIcon = /*#__PURE__*/ createIcon(FaGripLinesVertical)
export const FaGripVerticalIcon = /*#__PURE__*/ createIcon(FaGripVertical)
export const FaHourglassHalfIcon = /*#__PURE__*/ createIcon(FaHourglassHalf)
export const FaHouseIcon = /*#__PURE__*/ createIcon(FaHouse)
export const FaIdBadgeIcon = /*#__PURE__*/ createIcon(FaIdBadge)
export const FaIdCardIcon = /*#__PURE__*/ createIcon(FaIdCard)
export const FaImageIcon = /*#__PURE__*/ createIcon(FaImage)
export const FaInboxIcon = /*#__PURE__*/ createIcon(FaInbox)
export const FaIntercomIcon = /*#__PURE__*/ createIcon(FaIntercom)
export const FaKeyIcon = /*#__PURE__*/ createIcon(FaKey)
export const FaLeftLongIcon = /*#__PURE__*/ createIcon(FaLeftLong)
export const FaLeftRightIcon = /*#__PURE__*/ createIcon(FaLeftRight)
export const FaLightbulbIcon = /*#__PURE__*/ createIcon(FaLightbulb)
export const FaLinkIcon = /*#__PURE__*/ createIcon(FaLink)
export const FaListIcon = /*#__PURE__*/ createIcon(FaList)
export const FaListOlIcon = /*#__PURE__*/ createIcon(FaListOl)
export const FaListUlIcon = /*#__PURE__*/ createIcon(FaListUl)
export const FaLockIcon = /*#__PURE__*/ createIcon(FaLock)
export const FaLockOpenIcon = /*#__PURE__*/ createIcon(FaLockOpen)
export const FaMagnifyingGlassIcon = /*#__PURE__*/ createIcon(FaMagnifyingGlass)
export const FaMagnifyingGlassMinusIcon = /*#__PURE__*/ createIcon(FaMagnifyingGlassMinus)
export const FaMagnifyingGlassPlusIcon = /*#__PURE__*/ createIcon(FaMagnifyingGlassPlus)
export const FaMessageIcon = /*#__PURE__*/ createIcon(FaMessage)
export const FaMinusIcon = /*#__PURE__*/ createIcon(FaMinus)
export const FaMoneyCheckDollarIcon = /*#__PURE__*/ createIcon(FaMoneyCheckDollar)
export const FaPaintbrushIcon = /*#__PURE__*/ createIcon(FaPaintbrush)
export const FaPaperPlaneIcon = /*#__PURE__*/ createIcon(FaPaperPlane)
export const FaPaperclipIcon = /*#__PURE__*/ createIcon(FaPaperclip)
export const FaPenIcon = /*#__PURE__*/ createIcon(FaPen)
export const FaPenToSquareIcon = /*#__PURE__*/ createIcon(FaPenToSquare)
export const FaPencilIcon = /*#__PURE__*/ createIcon(FaPencil)
export const FaPiggyBankIcon = /*#__PURE__*/ createIcon(FaPiggyBank)
export const FaPlusIcon = /*#__PURE__*/ createIcon(FaPlus)
export const FaPowerOffIcon = /*#__PURE__*/ createIcon(FaPowerOff)
export const FaPrintIcon = /*#__PURE__*/ createIcon(FaPrint)
export const FaRegCircleQuestionIcon = /*#__PURE__*/ createIcon(FaRegCircleQuestion)
export const FaRegLightbulbIcon = /*#__PURE__*/ createIcon(FaRegLightbulb)
export const FaReplyIcon = /*#__PURE__*/ createIcon(FaReply)
export const FaRightFromBracketIcon = /*#__PURE__*/ createIcon(FaRightFromBracket)
export const FaRightLeftIcon = /*#__PURE__*/ createIcon(FaRightLeft)
export const FaRightLongIcon = /*#__PURE__*/ createIcon(FaRightLong)
export const FaRightToBracketIcon = /*#__PURE__*/ createIcon(FaRightToBracket)
export const FaRobotIcon = /*#__PURE__*/ createIcon(FaRobot)
export const FaRocketIcon = /*#__PURE__*/ createIcon(FaRocket)
export const FaRotateIcon = /*#__PURE__*/ createIcon(FaRotate)
export const FaRotateLeftIcon = /*#__PURE__*/ createIcon(FaRotateLeft)
export const FaShareNodesIcon = /*#__PURE__*/ createIcon(FaShareNodes)
export const FaShieldHalvedIcon = /*#__PURE__*/ createIcon(FaShieldHalved)
export const FaSlidersIcon = /*#__PURE__*/ createIcon(FaSliders)
export const FaSortDownIcon = /*#__PURE__*/ createIcon(FaSortDown)
export const FaSortIcon = /*#__PURE__*/ createIcon(FaSort)
export const FaSortUpIcon = /*#__PURE__*/ createIcon(FaSortUp)
export const FaSquareCheckIcon = /*#__PURE__*/ createIcon(FaSquareCheck)
export const FaSquareMinusIcon = /*#__PURE__*/ createIcon(FaSquareMinus)
export const FaSquarePlusIcon = /*#__PURE__*/ createIcon(FaSquarePlus)
export const FaSquarePollVerticalIcon = /*#__PURE__*/ createIcon(FaSquarePollVertical)
export const FaStarIcon = /*#__PURE__*/ createIcon(FaStar)
export const FaSuitcaseMedicalIcon = /*#__PURE__*/ createIcon(FaSuitcaseMedical)
export const FaTableIcon = /*#__PURE__*/ createIcon(FaTable)
export const FaTableListIcon = /*#__PURE__*/ createIcon(FaTableList)
export const FaTableColumnsIcon = /*#__PURE__*/ createIcon(FaTableColumns)
export const FaTagIcon = /*#__PURE__*/ createIcon(FaTag)
export const FaTagsIcon = /*#__PURE__*/ createIcon(FaTags)
export const FaThumbtackIcon = /*#__PURE__*/ createIcon(FaThumbtack)
export const FaToolboxIcon = /*#__PURE__*/ createIcon(FaToolbox)
export const FaTrashCanIcon = /*#__PURE__*/ createIcon(FaTrashCan)
export const FaTrashIcon = /*#__PURE__*/ createIcon(FaTrash)
export const FaTriangleExclamationIcon = /*#__PURE__*/ createIcon(FaTriangleExclamation)
export const FaTruckIcon = /*#__PURE__*/ createIcon(FaTruck)
export const FaTruckRampBoxIcon = /*#__PURE__*/ createIcon(FaTruckRampBox)
export const FaUpDownIcon = /*#__PURE__*/ createIcon(FaUpDown)
export const FaUpRightAndDownLeftFromCenterIcon = /*#__PURE__*/ createIcon(
  FaUpRightAndDownLeftFromCenter,
)
export const FaUpRightFromSquareIcon = /*#__PURE__*/ createIcon(FaUpRightFromSquare)
export const FaUserCheckIcon = /*#__PURE__*/ createIcon(FaUserCheck)
export const FaUserClockIcon = /*#__PURE__*/ createIcon(FaUserClock)
export const FaUserIcon = /*#__PURE__*/ createIcon(FaUser)
export const FaUserLargeIcon = /*#__PURE__*/ createIcon(FaUserLarge)
export const FaUserPenIcon = /*#__PURE__*/ createIcon(FaUserPen)
export const FaUserPlusIcon = /*#__PURE__*/ createIcon(FaUserPlus)
export const FaUsersIcon = /*#__PURE__*/ createIcon(FaUsers)
export const FaWindowMaximizeIcon = /*#__PURE__*/ createIcon(FaWindowMaximize)
export const FaWindowMinimizeIcon = /*#__PURE__*/ createIcon(FaWindowMinimize)
export const FaWindowRestoreIcon = /*#__PURE__*/ createIcon(FaWindowRestore)
export const FaXmarkIcon = /*#__PURE__*/ createIcon(FaXmark)
export const FaYenSignIcon = /*#__PURE__*/ createIcon(FaYenSign)

export { WarningIcon } from './WarningIcon'
export { SparklesIcon } from './SparklesIcon'
export { LanguageIcon } from './LanguageIcon'
