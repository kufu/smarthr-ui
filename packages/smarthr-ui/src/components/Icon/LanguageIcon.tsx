import React from 'react'
import { IconBase } from 'react-icons'

import { generateIcon } from './generateIcon'

export const LanguageIcon = /*#__PURE__*/ generateIcon((props) => (
  <IconBase {...props} viewBox="0 0 16 16">
    <path
      fillRule="evenodd"
      d="M14.5 8c0 .606-.083 1.193-.238 1.75h-2.206c.066-.563.1-1.149.1-1.75s-.034-1.187-.1-1.75h2.206c.155.557.238 1.144.238 1.75Zm-3.955-1.75c.072.554.111 1.14.111 1.75 0 .61-.04 1.196-.111 1.75H5.424A13.58 13.58 0 0 1 5.312 8c0-.61.04-1.196.112-1.75h5.12Zm1.253-1.5h1.832a6.529 6.529 0 0 0-2.802-2.604c.408.729.74 1.613.97 2.604ZM7.988 1.5c.312.002.959.276 1.616 1.537.253.485.475 1.063.648 1.713H5.717c.173-.65.394-1.228.647-1.713.656-1.258 1.3-1.533 1.614-1.537h.01ZM7.968 0a8 8 0 0 0 0 16H8a8 8 0 1 0-.033-16Zm2.86 13.854a6.528 6.528 0 0 0 2.802-2.604h-1.832c-.23.99-.562 1.875-.97 2.604Zm-.576-2.604H5.717c.173.65.394 1.228.647 1.713.656 1.258 1.3 1.533 1.614 1.537h.01c.312-.002.959-.275 1.616-1.537a8.418 8.418 0 0 0 .648-1.713Zm-5.123 2.583c-.402-.725-.73-1.602-.958-2.583H2.37a6.53 6.53 0 0 0 2.76 2.583ZM3.913 9.75H1.738A6.506 6.506 0 0 1 1.5 8c0-.606.083-1.193.238-1.75h2.175a15.158 15.158 0 0 0 0 3.5Zm.258-5H2.37a6.53 6.53 0 0 1 2.76-2.583c-.403.725-.731 1.602-.959 2.583Z"
    />
  </IconBase>
))
