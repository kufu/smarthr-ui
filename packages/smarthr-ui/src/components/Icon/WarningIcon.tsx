import React from 'react'
import { IconBase } from 'react-icons'

import { generateIcon } from './generateIcon'

export const WarningIcon = /*#__PURE__*/ generateIcon((props) => (
  <IconBase {...props} viewBox="0 0 16 16">
    <path
      d="m8.863 1.745 6.75 11.5a.998.998 0 0 1-.862 1.505H1.25c-.358 0-.69-.193-.868-.502a1.005 1.005 0 0 1 .005-1.003l6.75-11.5a.998.998 0 0 1 1.725 0Z"
      className="smarthr-ui-WarningIcon-wrapper shr-fill-warning-yellow shr-stroke-black shr-stroke-0.5"
    />
    <path
      className="smarthr-ui-WarningIcon-mark shr-fill-black"
      d="M8 5a.748.748 0 0 0-.75.75v3.5c0 .416.334.75.75.75s.75-.334.75-.75v-3.5A.748.748 0 0 0 8 5Zm1 7a1 1 0 1 0-2 0 1 1 0 0 0 2 0Z"
    />
  </IconBase>
))
