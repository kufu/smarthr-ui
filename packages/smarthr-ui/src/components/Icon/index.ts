export * from './Icon'
export * from './DeprecatedIcon'
export { generateIcon, Props as ComponentProps } from './generateIcon'
