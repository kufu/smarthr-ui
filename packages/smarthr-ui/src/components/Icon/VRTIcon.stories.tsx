import { StoryFn } from '@storybook/react'
import * as React from 'react'
import styled from 'styled-components'

import { InformationPanel } from '../InformationPanel'

import * as Icons from './Icon'
import { All, Color } from './Icon.stories'

const { FaAddressCardIcon } = Icons

export default {
  title: 'Media（メディア）/Icon',
  component: FaAddressCardIcon,
  parameters: {
    withTheming: true,
  },
}

export const VRTForcedColors: StoryFn = () => (
  <>
    <VRTInformationPanel title="VRT 用の Story です">
      Chromatic 上では強制カラーモードで表示されます
    </VRTInformationPanel>
    <Title>Color</Title>
    <VRTColorWrapper>
      <Color />
    </VRTColorWrapper>
    <Title>All</Title>
    <All />
  </>
)
VRTForcedColors.parameters = {
  chromatic: { forcedColors: 'active' },
}

const VRTInformationPanel = styled(InformationPanel)`
  margin-bottom: 24px;
`
const Title = styled.p`
  margin: 0 0 16px;
`
const VRTColorWrapper = styled.div`
  padding: 0 16px 16px 0;
`
