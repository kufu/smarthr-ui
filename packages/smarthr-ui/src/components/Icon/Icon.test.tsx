import React from 'react'
import { create } from 'react-test-renderer'

import { FaAddressCardIcon } from './Icon'

const Icon = () => (
  <>
    <span id="text">連絡帳</span>
    <FaAddressCardIcon />
    <FaAddressCardIcon alt="連絡帳" />
    <FaAddressCardIcon aria-label="連絡帳" />
    <FaAddressCardIcon aria-labelledby="text" />
  </>
)

describe('Icon', () => {
  it('should be match snapshot', () => {
    const testRenderer = create(<Icon />)
    expect(testRenderer.toJSON()).toMatchSnapshot()
  })
})
