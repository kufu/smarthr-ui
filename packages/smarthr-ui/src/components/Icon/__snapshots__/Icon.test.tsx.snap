// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon should be match snapshot 1`] = `
[
  <span
    id="text"
  >
    連絡帳
  </span>,
  <svg
    aria-hidden={true}
    className="smarthr-ui-Icon group-[]/iconWrapper:shr-shrink-0 group-[]/iconWrapper:shr-translate-y-[0.125em] forced-colors:shr-fill-[CanvasText]"
    fill="currentColor"
    focusable={false}
    height="1em"
    role="img"
    stroke="currentColor"
    strokeWidth="0"
    style={
      {
        "color": undefined,
      }
    }
    viewBox="0 0 576 512"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M64 32C28.7 32 0 60.7 0 96V416c0 35.3 28.7 64 64 64H512c35.3 0 64-28.7 64-64V96c0-35.3-28.7-64-64-64H64zm80 256h64c44.2 0 80 35.8 80 80c0 8.8-7.2 16-16 16H80c-8.8 0-16-7.2-16-16c0-44.2 35.8-80 80-80zm-32-96a64 64 0 1 1 128 0 64 64 0 1 1 -128 0zm256-32H496c8.8 0 16 7.2 16 16s-7.2 16-16 16H368c-8.8 0-16-7.2-16-16s7.2-16 16-16zm0 64H496c8.8 0 16 7.2 16 16s-7.2 16-16 16H368c-8.8 0-16-7.2-16-16s7.2-16 16-16zm0 64H496c8.8 0 16 7.2 16 16s-7.2 16-16 16H368c-8.8 0-16-7.2-16-16s7.2-16 16-16z"
    />
  </svg>,
  <span
    className="shr-absolute -shr-top-px shr-left-0 shr-h-px shr-w-px shr-overflow-hidden shr-whitespace-nowrap shr-border-0 shr-p-0 [clip-path:inset(100%)] [clip:rect(0_0_0_0)]"
  >
    連絡帳
  </span>,
  <svg
    aria-hidden={true}
    className="smarthr-ui-Icon group-[]/iconWrapper:shr-shrink-0 group-[]/iconWrapper:shr-translate-y-[0.125em] forced-colors:shr-fill-[CanvasText]"
    fill="currentColor"
    focusable={false}
    height="1em"
    role="img"
    stroke="currentColor"
    strokeWidth="0"
    style={
      {
        "color": undefined,
      }
    }
    viewBox="0 0 576 512"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M64 32C28.7 32 0 60.7 0 96V416c0 35.3 28.7 64 64 64H512c35.3 0 64-28.7 64-64V96c0-35.3-28.7-64-64-64H64zm80 256h64c44.2 0 80 35.8 80 80c0 8.8-7.2 16-16 16H80c-8.8 0-16-7.2-16-16c0-44.2 35.8-80 80-80zm-32-96a64 64 0 1 1 128 0 64 64 0 1 1 -128 0zm256-32H496c8.8 0 16 7.2 16 16s-7.2 16-16 16H368c-8.8 0-16-7.2-16-16s7.2-16 16-16zm0 64H496c8.8 0 16 7.2 16 16s-7.2 16-16 16H368c-8.8 0-16-7.2-16-16s7.2-16 16-16zm0 64H496c8.8 0 16 7.2 16 16s-7.2 16-16 16H368c-8.8 0-16-7.2-16-16s7.2-16 16-16z"
    />
  </svg>,
  <svg
    aria-label="連絡帳"
    className="smarthr-ui-Icon group-[]/iconWrapper:shr-shrink-0 group-[]/iconWrapper:shr-translate-y-[0.125em] forced-colors:shr-fill-[CanvasText]"
    fill="currentColor"
    focusable={false}
    height="1em"
    role="img"
    stroke="currentColor"
    strokeWidth="0"
    style={
      {
        "color": undefined,
      }
    }
    viewBox="0 0 576 512"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M64 32C28.7 32 0 60.7 0 96V416c0 35.3 28.7 64 64 64H512c35.3 0 64-28.7 64-64V96c0-35.3-28.7-64-64-64H64zm80 256h64c44.2 0 80 35.8 80 80c0 8.8-7.2 16-16 16H80c-8.8 0-16-7.2-16-16c0-44.2 35.8-80 80-80zm-32-96a64 64 0 1 1 128 0 64 64 0 1 1 -128 0zm256-32H496c8.8 0 16 7.2 16 16s-7.2 16-16 16H368c-8.8 0-16-7.2-16-16s7.2-16 16-16zm0 64H496c8.8 0 16 7.2 16 16s-7.2 16-16 16H368c-8.8 0-16-7.2-16-16s7.2-16 16-16zm0 64H496c8.8 0 16 7.2 16 16s-7.2 16-16 16H368c-8.8 0-16-7.2-16-16s7.2-16 16-16z"
    />
  </svg>,
  <svg
    aria-labelledby="text"
    className="smarthr-ui-Icon group-[]/iconWrapper:shr-shrink-0 group-[]/iconWrapper:shr-translate-y-[0.125em] forced-colors:shr-fill-[CanvasText]"
    fill="currentColor"
    focusable={false}
    height="1em"
    role="img"
    stroke="currentColor"
    strokeWidth="0"
    style={
      {
        "color": undefined,
      }
    }
    viewBox="0 0 576 512"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M64 32C28.7 32 0 60.7 0 96V416c0 35.3 28.7 64 64 64H512c35.3 0 64-28.7 64-64V96c0-35.3-28.7-64-64-64H64zm80 256h64c44.2 0 80 35.8 80 80c0 8.8-7.2 16-16 16H80c-8.8 0-16-7.2-16-16c0-44.2 35.8-80 80-80zm-32-96a64 64 0 1 1 128 0 64 64 0 1 1 -128 0zm256-32H496c8.8 0 16 7.2 16 16s-7.2 16-16 16H368c-8.8 0-16-7.2-16-16s7.2-16 16-16zm0 64H496c8.8 0 16 7.2 16 16s-7.2 16-16 16H368c-8.8 0-16-7.2-16-16s7.2-16 16-16zm0 64H496c8.8 0 16 7.2 16 16s-7.2 16-16 16H368c-8.8 0-16-7.2-16-16s7.2-16 16-16z"
    />
  </svg>,
]
`;
